repos/spack_repo/builtin/packages/py_hist/package.py

```diff
@@ -13,23 +13,53 @@ class PyHist(PythonPackage):
     homepage = "https://github.com/scikit-hep/hist"
     pypi = "hist/hist-2.5.2.tar.gz"
 
+    maintainers("wdconinc")
+
     license("BSD-3-Clause")
 
+    version("2.8.1", sha256="7da7c900e2ef6d425793da1a9adac424ebc013a8eabf29b24301f70898218d9d")
+    version("2.7.3", sha256="f9f9b56809b190bb546698789cc0d7d040934fc5141d2763c6e49d65e81dbc0b")
+    version("2.6.3", sha256="dede097733d50b273af9f67386e6dcccaab77e900ae702e1a9408a856e217ce9")
     version("2.6.1", sha256="ee9034795fd2feefed923461aaccaf76f87c1f8d5414b1e704faa293ceb4fc27")
     version("2.5.2", sha256="0bafb8b956cc041f1b26e8f5663fb8d3b8f7673f56336facb84d8cfdc30ae2cf")
 
     variant("plot", default=False, description="Add support for drawing histograms")
+    variant("dask", default=False, description="Add support for dask histograms", when="@2.6.3:")
+    variant("fit", default=False, description="Add support for fitting histograms", when="@2.7.1:")
 
     depends_on("python@3.7:", type=("build", "run"))
-    depends_on("py-setuptools@45:", type="build")
-    depends_on("py-setuptools-scm@3.4:+toml", type="build")
+    depends_on("python@3.8:", type=("build", "run"), when="@2.8.0:")
+    with when("@:2.6.1"):
+        depends_on("py-setuptools@45:", type="build")
+        depends_on("py-setuptools-scm@3.4:+toml", type="build")
+    with when("@2.6.2:"):
+        depends_on("py-hatchling", type="build")
+        depends_on("py-hatch-vcs", type="build")
+
     depends_on("py-boost-histogram@1.2.0:1.2", when="@2.5.2", type=("build", "run"))
-    depends_on("py-boost-histogram@1.3.1:1.3", when="@2.6.1", type=("build", "run"))
+    depends_on("py-boost-histogram@1.3.1:1.3", when="@2.6.1:2.7.1", type=("build", "run"))
+    depends_on("py-boost-histogram@1.3.1:1.4", when="@2.7.2:", type=("build", "run"))
+    depends_on("py-boost-histogram@1.3.1:1.5", when="@2.8.0:", type=("build", "run"))
     depends_on("py-histoprint@2.2.0:", type=("build", "run"))
-    depends_on("py-numpy@1.14.5:", type=("build", "run"))
-    depends_on("py-typing-extensions@3.7:", when="^python@:3.7", type=("build", "run"))
+    depends_on("py-numpy@1.14.5:", type=("build", "run"), when="@:2.7.1,2.7.3:")
+    depends_on("py-numpy@1.14.5:", type=("build", "run"), when="@2.7.2 ^python@:3.11")
+    depends_on("py-numpy@1.26:", type=("build", "run"), when="@2.7.2 ^python@3.12:")
+    depends_on("py-typing-extensions@3.7:", when="@:2.6 ^python@:3.7", type=("build", "run"))
+    depends_on("py-typing-extensions@4:", when="@2.7: ^python@:3.11", type=("build", "run"))
```

Comment:
The range is inclusive so the cap on `python` should be `3.10`, not `3.11`:

```suggestion
    depends_on("py-typing-extensions@4:", when="@2.7: ^python@:3.10", type=("build", "run"))
```